include/seastar/core/sstring.hh

```diff
@@ -592,18 +592,26 @@ public:
 template <typename char_type, typename Size, Size max_size, bool NulTerminate>
 constexpr Size basic_sstring<char_type, Size, max_size, NulTerminate>::npos;
 
+namespace internal {
+template <class T> struct is_sstring : std::false_type {};
+template <typename char_type, typename Size, Size max_size, bool NulTerminate>
+struct is_sstring<basic_sstring<char_type, Size, max_size, NulTerminate>> : std::true_type {};
+}
+
 template <typename string_type = sstring>
 string_type uninitialized_string(size_t size) {
-    string_type ret;
-    // FIXME: use __resize_default_init if available
-    ret.resize(size);
-    return ret;
+    if constexpr (internal::is_sstring<string_type>::value) {
+        return string_type(typename string_type::initialized_later(), size);
+    } else {
+        string_type ret;
+        // FIXME: use __resize_default_init if available
+        ret.resize(size);
+        return ret;
+    }
 }
 
 template <typename char_type, typename Size, Size max_size, bool NulTerminate>
 basic_sstring<char_type, Size, max_size, NulTerminate> uninitialized_string(size_t size) {
-    using sstring_type = basic_sstring<char_type, Size, max_size, NulTerminate>;
-    return sstring_type(sstring_type::initialized_later(), size);
```

Comment:
How does this work? A function with no return?

Reply:
ahh, i forgot to remove this function.